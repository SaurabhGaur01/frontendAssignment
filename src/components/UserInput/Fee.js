import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setFee } from '../../ducks/fee';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const Fee = ({
    actionSetFee,
    fee, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputFee = parseInt(event.target.value);
        actionSetFee(inputFee);
    }

    return (
        <div>
            Fee:
            <input id= "fee" type="text" name="fee" defaultValue={fee} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

Fee.propTypes = {
    actionSetFee: PropTypes.func.isRequired,
    fee: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetFee: setFee,
}

export const mapStateToProps = state => ({
    fee: state.fee,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { Fee as TestableFee };

export default hocChain(Fee);
