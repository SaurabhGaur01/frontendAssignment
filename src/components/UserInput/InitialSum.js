import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setInitialSum } from '../../ducks/initialSum';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const InitialSum = ({
    actionSetInitialSum,
    initialSum, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputInitialSum = parseInt(event.target.value);
        actionSetInitialSum(inputInitialSum);
    }

    return (
        <div>
            Initial Sum:
            <input id= "initialSum" type="text" name="initialSum" defaultValue={initialSum} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

InitialSum.propTypes = {
    actionSetInitialSum: PropTypes.func.isRequired,
    initialSum: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetInitialSum: setInitialSum,
}

export const mapStateToProps = state => ({
    initialSum: state.initialSum,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { InitialSum as TestableInitialSum };

export default hocChain(InitialSum);
