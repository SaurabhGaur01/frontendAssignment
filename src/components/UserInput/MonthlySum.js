import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setMonthlySum } from '../../ducks/monthlySum';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const MonthlySum = ({
    actionSetMonthlySum,
    monthlySum, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputMonthlySum = parseInt(event.target.value);
        actionSetMonthlySum(inputMonthlySum);
    }

    return (
        <div>
            Monthly Sum:
            <input type="text" id="monthlySum" name="monthlySum" defaultValue={monthlySum} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

MonthlySum.propTypes = {
    actionSetMonthlySum: PropTypes.func.isRequired,
    monthlySum: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetMonthlySum: setMonthlySum,
}

export const mapStateToProps = state => ({
    monthlySum: state.monthlySum,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { MonthlySum as TestableMonthlySum };

export default hocChain(MonthlySum);
