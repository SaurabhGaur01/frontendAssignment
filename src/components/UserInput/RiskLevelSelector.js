import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setRiskLevel } from '../../ducks/riskLevel';

const maxRiskLevel = 25;
const defaultRisk = 10;

const RiskLevelSelector = ({
    actionSetRiskLevel, 
}) => {
    const onChange = (event) => {
        var inputRiskLevel = parseInt(event.target.value);
        actionSetRiskLevel(inputRiskLevel);
    }

    const options = [];
    for(let k=1; k<=maxRiskLevel; ++k) {
        options.push(
            <option key={k} value={k}>{k}</option>
        );
    }

    return (
        <div>
            Risk level:
            <select id="riskLevel" className="riskLevel" onChange={onChange} defaultValue={defaultRisk}>
                {options}
            </select>
        </div>
    );
}

RiskLevelSelector.propTypes = {
    actionSetRiskLevel: PropTypes.func.isRequired,
};

const mapDispatchAsProps = {
    actionSetRiskLevel: setRiskLevel,
}

const hocChain = compose(
    connect(null, mapDispatchAsProps),
);

export { RiskLevelSelector as TestableRiskLevelSelector };

export default hocChain(RiskLevelSelector);
