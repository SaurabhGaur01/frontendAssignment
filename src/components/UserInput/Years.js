import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setYears } from '../../ducks/years';

const useStyles = makeStyles({
    inputWidth: {
      width: '45px',
    },
});

const Years = ({
    actionSetYears,
    years, 
}) => {
    const classes = useStyles();
    const onChange = (event) => {
        var inputYears = parseInt(event.target.value);
        actionSetYears(inputYears);
    }

    return (
        <div>
            Years:
            <input type="text" id="years" name="years" defaultValue={years} onChange={onChange} className={classes.inputWidth} />
        </div>
    );
}

Years.propTypes = {
    actionSetYears: PropTypes.func.isRequired,
    years: PropTypes.number.isRequired,
};

const mapDispatchAsProps = {
    actionSetYears: setYears,
}

export const mapStateToProps = state => ({
    years: state.years,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { Years as TestableYears };

export default hocChain(Years);
