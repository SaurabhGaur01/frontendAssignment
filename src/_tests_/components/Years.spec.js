import React from 'react';
import { shallow } from 'enzyme';
import { TestableYears, mapStateToProps } from '../../components/UserInput/Years';

describe('Years Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionSetYears: jest.fn(),
            years: 10,
        };
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableYears {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('onChange', () => {
        it('should call actionSetYears when user input any value', () => {
            const wrapper = shallow(<TestableYears {...props} />);
            const event = { target: { value: 10}};
            wrapper.find('#years').simulate('change', event);
            expect(props.actionSetYears).toHaveBeenCalledTimes(1);
            expect(props.actionSetYears).toHaveBeenCalledWith(10);
        })
    })

    describe('mapStateToProps', () => {
        const yearsMock = 'mockValue';
    
        const mockedState = {
            years: yearsMock,
        };

        it('should return the module state from the redux store', () => {
            const mockPropsFromRedux = mapStateToProps(mockedState);
            expect(mockPropsFromRedux).toEqual(
                {
                    years: yearsMock,
                }
            )
        })
    })
})