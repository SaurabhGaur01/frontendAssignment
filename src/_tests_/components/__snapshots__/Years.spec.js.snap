// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Years Component should render correctly 1`] = `
<div>
  Years:
  <input
    className="makeStyles-inputWidth-1"
    defaultValue={10}
    id="years"
    name="years"
    onChange={[Function]}
    type="text"
  />
</div>
`;
